plugin-classloader/src/main/java/org/apache/atlas/plugin/classloader/AtlasPluginClassLoader.java

```diff
@@ -206,7 +209,7 @@ public void activate() {
         if (LOG.isDebugEnabled()) {
             LOG.debug("==> AtlasPluginClassLoader.activate()");
         }
-
+        previousLoader = Thread.currentThread().getContextClassLoader();
```

Comment:
AtlasPluginClassLoader instance could be used in multiple threads simultaneously. So, saving 'previousLoader' in an instance member is not a safe option. @rmani (Ramesh Mani) is working on addressing the same issue in Ranger via RANGER-2376. I would suggest to wait for his Ranger fix and apply the same in Atlas as well.

Reply:
I wonder if it would be safe to save it on a per thread basis then?

Reply:
Or just handling this in the HiveHook it self? Then it should all be in the same thread and would be hive specific.